core/vdbe/mod.rs

```diff
@@ -512,14 +512,19 @@ impl Program {
                     lhs,
                     rhs,
                     target_pc,
+                    jump_if_null,
                 } => {
                     assert!(target_pc.is_offset());
                     let lhs = *lhs;
                     let rhs = *rhs;
                     let target_pc = *target_pc;
                     match (&state.registers[lhs], &state.registers[rhs]) {
                         (_, OwnedValue::Null) | (OwnedValue::Null, _) => {
-                            state.pc += 1;
+                            if *jump_if_null {
```

Comment:
@jussisaurio 

Why not change ParrtialEq/PartialCmp of OwnedValue to return false when 2 values are null. And then lets remove the special handling in our code for nulls and if false we step counter by 1.